move_group/src/default_capabilities/execute_service_capability.cpp

```diff
@@ -76,6 +76,8 @@ bool move_group::MoveGroupExecuteService::executeTrajectoryService(moveit_msgs::
         res.error_code.val = moveit_msgs::MoveItErrorCodes::TIMED_OUT;
       else
         res.error_code.val = moveit_msgs::MoveItErrorCodes::CONTROL_FAILED;
+      // wait for all planning scene updates to be processed
+      context_->planning_scene_monitor_->syncSceneUpdates();
```

Comment:
I don't see any reason to sync the scene here. Did you have a something use-case in mind?